plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLScriptParser.java

```diff
@@ -107,7 +107,8 @@ private static SQLScriptElement parseQueryImpl(
 
         // Parse range
         TPRuleBasedScanner ruleScanner = context.getScanner();
-        boolean useBlankLines = !scriptMode && context.getSyntaxManager().getStatementDelimiterMode().useBlankLine;
+        boolean useBlankLines = (!scriptMode && context.getSyntaxManager().getStatementDelimiterMode().useBlankLine)
+            || (scriptMode && context.getSyntaxManager().getStatementDelimiterMode().useSmart);
```

Comment:
It's REALLY hard to tread. It can be simplified:

```suggestion
        boolean useBlankLines = scriptMode
            ? context.getSyntaxManager().getStatementDelimiterMode().useSmart
            : context.getSyntaxManager().getStatementDelimiterMode().useBlankLine;
```

Reply:
it loses sense this way